rita_common/src/tunnel_manager/contact_peers.rs

```diff
@@ -111,104 +186,64 @@ pub async fn tm_contact_peers(pl: PeerListener) {
+                    // it will just fill the logs with failed dns resolution attempts or result
+                    // in bad behavior, we do allow the addressing of direct ip address gateways
+                    // for the special case that the user is attempting some special behavior
+                    if is_gateway {
+                        manual_peers_dns_fut
+                            .push(tm_neighbor_inquiry_hostname(manual_peer.to_string()));
                     }
                 }
             }
         }
     }
-}
-
-/// Sets out to contacts a local mesh neighbor, takes a speculative port (only assigned if the neighbor
-/// responds successfully). This function is used for mesh peers
-fn contact_neighbor(
-    peer: &Peer,
-    our_port: u16,
-    socket: &UdpSocket,
-    send_addr: SocketAddr,
-) -> Result<(), RitaCommonError> {
-    let new_msg = NewHello {
-        my_id: LocalIdentity {
-            global: settings::get_rita_common().get_identity().ok_or_else(|| {
-                RitaCommonError::MiscStringError("Identity has no mesh IP ready yet".to_string())
-            })?,
-            wg_port: our_port,
-            have_tunnel: None,
-        },
-        to: *peer,
-        response: false,
-    };
-
-    // new send_hello call using udp socket
-    // We do not need the old http hello except for exits, which are called as manual peers
-    send_hello(&new_msg, socket, send_addr, our_port);
-
-    Ok(())
-}
 
-/// Uses Hello Handler to send a Hello over http. Takes a speculative port (only assigned
-/// if neighbor responds successfully)
-async fn contact_manual_peer(peer: &Peer, our_port: u16) -> Result<(), RitaCommonError> {
-    let mut settings = settings::get_rita_common();
-    let changed = KI.manual_peers_route(
-        &peer.contact_socket.ip(),
-        &mut settings.network.last_default_route,
-    )?;
-
-    let msg = Hello {
-        my_id: LocalIdentity {
-            global: settings.get_identity().ok_or_else(|| {
-                RitaCommonError::MiscStringError("Identity has no mesh IP ready yet".to_string())
-            })?,
-            wg_port: our_port,
-            have_tunnel: None,
-        },
-        to: *peer,
-    };
-
-    if changed {
-        settings::set_rita_common(settings);
+    // now that we've constructed our futures arrays, now we execute them in parallel
+    // executing udp peers in this way has no advantage of doing them sync becuase it's all
+    // sync actions, but for manual peers the advantage is huge since there are half a dozen exits
+    // and they each may take seconds to respond, joining the udp peers in just lets us run their sync
+    // operations while efficiencly waiting for http exit responses
+    let a = join_all(udp_peers_fut);
+    let b = join_all(manual_peers_ip_fut);
+    let c = join_all(manual_peers_dns_fut);
+    let (udp_result, manual_result_ip, manual_result_dns) = join3(a, b, c).await;
```

Comment:
This looks fine, but out of curiosity, would it be possible to add all 3 vecs into one fut vector and run join_all(fut)?